### ISSUE TITLE
relative_files & include broken since 7.0

### ISSUE THREAD

**Issue by ikonst** [coveragepy/coveragepy#1604]
**Describe the bug**

1. Create simple test case with `relative_files = true`
   ```shell
   mkdir foo
   echo 'print()" > foo/bar.py
   echo "[run]\nrelative_files = true" > .coveragerc
   ```

2. `coverage run foo/bar.py`
3. `coverage report --include=foo/bar.py`

**Expected (as behavior was before 7.0):**
```
Name         Stmts   Miss  Cover
--------------------------------
foo/bar.py       1      0   100%
--------------------------------
TOTAL            1      0   100%
```

**Actual:**
```
No data to report
```

**Additional analysis:**

It appears that the regex created from the `foo/bar.py` glob pattern includes the current working directory, which might explain why it wouldn't match against relative paths anymore.

**Comment by nedbat** [comment]
Hmm, I see what you mean.  It will still work if you use `--include='*/foo/bar.py'`.

**Comment by ikonst** [comment]
BTW, regarding `--include='*/foo/bar.py'`:

1. Would that get me more than I bargained for, e.g. also `spam/foo/bar.py`?
2. Is it compatible with 6.x?

**Comment by nedbat** [comment]
I have a possible fix, but it's hard to work through all the permutations.  I will investigate.

**Comment by nedbat** [comment]
@ikonst do you need to use the same settings file for both coverage 6.x and 7.x?

**Comment by ikonst** [comment]
Preferably, since I have a script invoke that report across repositories where different versions might be used. Also, I think the current syntax is intuitive for the relative files case.

**Comment by marcgibbons** [comment]
@marcgibbons at PyCon 23

**Comment by marcgibbons** [comment]
Also, looks like #1571 is a duplicate.

**Comment by nedbat** [comment]
This is fixed in ee6506f0.

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.2.4](https://pypi.org/project/coverage/7.2.4).
